rcl/test/rcl/test_security_directory.cpp

```diff
@@ -0,0 +1,141 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <string>
+#include "rcl/security_directory.h"
+#include "rcutils/filesystem.h"
+
+#define ROOT_NAMESPACE "/"
+#define TEST_SECURITY_DIRECTORY_RESOURCES_DIR_NAME "test_security_directory"
+#define TEST_NODE_NAME "dummy_node"
+#define TEST_NODE_NAMESPACE ROOT_NAMESPACE TEST_SECURITY_DIRECTORY_RESOURCES_DIR_NAME
+
+char g_envstring[512] = {0};
+
+static int putenv_wrapper(const char * env_var)
+{
+#ifdef _WIN32
+  return _putenv(env_var);
+#else
+  return putenv(reinterpret_cast<char *>(const_cast<char *>(env_var)));
+#endif
+}
+
+TEST(test_rcl_get_secure_root, failureScenarios) {
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+  /* Before setting the environment variable */
```

Comment:
I don't think we can guarantee that the environment variable hasn't been set before this test is run. Since we [can't assume the tests are run in any particular order](https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#sharing-resources-between-tests-in-the-same-test-suite), it's possible the environment is set by another test. I think we should explicitly unset the variable, or set it to the empty string (whatever works), before the assertions in this test.

Alternatively, we can add a [test fixture](https://github.com/google/googletest/blob/master/googletest/docs/primer.md#test-fixtures-using-the-same-data-configuration-for-multiple-tests) and make use of the `setUp()` method to reset the common environment variables used between tests.

Reply:
very good point, will unset variables before each test